lib/connections/quic_dial.go

```diff
@@ -80,7 +80,7 @@ func (d *quicDialer) Dial(ctx context.Context, _ protocol.DeviceID, uri *url.URL
 	stream, err := session.OpenStreamSync(ctx)
 	if err != nil {
 		// It's ok to close these, this does not close the underlying packetConn.
-		_ = session.Close()
+		_ = session.CloseWithError(0, err.Error())
```

Comment:
Probably doesn't matter, but I'd expect `1` for the generic BEP error (or just any generic error)? Same below.